flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java

```diff
@@ -548,9 +549,57 @@ public ServerMapping getMappingById(@PathVariable("id") String id) {
    @RequestMapping(value = "/mapping/hostnameip", method = RequestMethod.POST)
    public void createHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
       BaseDocumentUtil.generateID(mapping);
+      String ip = mapping.getIp();
+      if(ip == null || ip.isEmpty() || !ip.matches(HandleURL.IP_REGX)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid ip address",
+               null);
+      }
+      if (!assetService.isAssetNameValidate(mapping)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid asset name",
+               null);
+      }
       assetIPMappingRepository.save(mapping);
    }
 
+   @ResponseStatus(HttpStatus.OK)
+   @RequestMapping(value = "/mapping/hostnameip", method = RequestMethod.PUT)
+   public void updateHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
+      BaseDocumentUtil.generateID(mapping);
+      AssetIPMapping oldMapping = assetIPMappingRepository.findOne(mapping.getId());
+      if(!assetService.isAssetNameValidate(mapping)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR,"Invalid asset name",null);
+      }
+      oldMapping.setAssetname(mapping.getAssetname());
```

Comment:
check if name not changed then just return.

Reply:
Done